### PR TITLE
Added toggle for Friend Name overhead and added Hits/Prayer/Fatigue <…

### PR DESCRIPTION
**Pull request by Hubcapp**
…800pxwide overlay to Combat Menu toggle

The Combat Menu toggle should be temporary in favour of actually moving the Hits/Prayer/Fatigue overlay in response to Combat Menu being open, but for me this is an improvement.

### CODE REVIEW

File: src/Game/Renderer.java [hunk 1]
```diff
@@ -409,7 +409,34 @@ public static void present(Graphics g, Image image) {
 					drawShadowText(g2, Client.player_name, x, y, color_text, false);
 				y += 16;
 				drawShadowText(g2, "Region: (" + Client.regionX + "," + Client.regionY + ")", x, y, color_text, false);
+				/*
```
Comment (OrN) [inline src/Game/Renderer.java]: Why is all of this commented out?

File: src/Game/Renderer.java [hunk 2]
```diff
@@ -409,7 +409,34 @@ public static void present(Graphics g, Image image) {
...
+				y += 32;
```
Comment (OrN) [inline src/Game/Renderer.java]: Why is all of this commented out?

File: src/Game/Renderer.java [hunk 3]
```diff
@@ -409,7 +409,34 @@ public static void present(Graphics g, Image image) {
...
+				y += 16;
+				drawShadowText(g2, "show_sleeping: (" + Client.show_sleeping + ")", x, y, color_text, false);
+				y += 16;
+				drawShadowText(g2, "show_trade: (" + Client.show_trade + ")", x, y, color_text, false);
+				y += 16;
+				drawShadowText(g2, "show_tradeconfirm: (" + Client.show_tradeconfirm + ")", x, y, color_text, false);
+				y +=  16;
+				drawShadowText(g2, "show_welcome: (" + Client.show_welcome + ")", x, y, color_text, false);
+				y += 16;
+				/* */
```
Comment (OrN) [inline src/Game/Renderer.java]: Why is all of this commented out?

File: src/Client/JClassPatcher.java [hunk 4]
```diff
@@ -144,6 +144,7 @@ private void patchGeneric(ClassNode node) {
 			hookClassVariable(methodNode, "client", "qd", "I", "Game/Camera", "fov", "I", false, true);
 
 			hookClassVariable(methodNode, "client", "Fe", "Z", "Game/Client", "show_bank", "Z", true, false);
+			hookClassVariable(methodNode, "client", "Fe", "Z", "Game/Client", "show_combat", "Z", true, false); //this won't work
```
Comment (OrN) [inline src/Client/JClassPatcher.java]: Why are we hooking what appears to be the bank interface being drawn as _show_combat_, can _show_bank_ be used instead?

**Comment by OrN** [comment]
When those issues are fixed, looks good to me

File: src/Game/Client.java [hunk 5]
```diff
@@ -465,6 +467,7 @@ private static void dumpStrings() {
 	public static long magic_timer = 0L;
 
 	public static boolean show_bank;
+	public static boolean show_combat;
```
Comment (OrN) [inline src/Game/Client.java]: Same thing here, show_combat is hooking the same variable as show_bank

**Comment by OrN** [comment]
When those issues are fixed, looks good to me

**Comment by Hubcapp** [comment]
Do you know what class is the combat class? To fix it, I'd need to find a
"player in combat"  bool.

**Comment by OrN** [comment]
Not right off, I'll see if I can find a way to detect if the player is in combat

**Comment by OrN** [comment]
Just pushed a commit to grab combat timer for logout, the function is Client.isInCombat()

I didn't test it since I don't have membership, so tell me if it works properly.

**Comment by Hubcapp** [comment]
I will when I get home from protesting hillary, thanks.

**Comment by OrN** [comment]
I went ahead and pushed your change to master, tell me if it works properly

**Comment by Hubcapp** [comment]
Preston - -

Okay cool! I'm at the amusement Park with friends now, and I'm gonna put in
a couple more toggles before I'm done working on it. (was gonna make a few
more changes before Making a new pull request)

Thanks for accepting my changes. :P

- -Tyler

**Comment by OrN** [comment]
Alright cool, I'll close this one for now and when you add more or want to change it just go ahead and open a new pull request

**Pull request closed without merging**